### PR TITLE
Added shift to PID output, added Talon SRX encoders

### PR DESCRIPTION
**Pull request by EthanM827**

### CODE REVIEW

File: src/main/java/frc/robot/commands/pidcommands/MoveUsingEncoder.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
   public MoveUsingEncoder(DriveTrain drivetrain, double distance) {
     super(
         // The controller that the command will use
-        new PIDController(0, 0, 0),
+        new PIDController4905(0, 0, 0, 0),
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/commands/pidcommands/MoveUsingEncoder.java]: I spy magic numbers that could be configured

File: src/main/java/frc/robot/pidcontroller/PIDController4905.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  public PIDController4905(double Kp, double Ki, double Kd, double minOutputToMove) {
+    super(Kp, Ki, Kd);
+    m_minOutputToMove = minOutputToMove;
+  }
+
+  @Override
+  public double calculate(double measurement) {
+    double output = super.calculate(measurement);
+    output = output * (1 - m_minOutputToMove) + m_minOutputToMove;
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/pidcontroller/PIDController4905.java]: what's the documentation for this equation?

File: src/main/java/frc/robot/pidcontroller/PIDControllerProposed.java [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+    if (m_Ki != 0) {
+      m_totalError = MathUtil.clamp(m_totalError + m_positionError * m_period, m_minimumIntegral / m_Ki,
+          m_maximumIntegral / m_Ki);
+    }
+
+    m_pError = m_Kp * m_positionError;
+    m_iError = m_Ki * m_totalError;
+    m_dError = m_Kd * m_velocityError;
+
+    return m_pError + m_iError + m_dError;
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/pidcontroller/PIDControllerProposed.java]: What's the documentation for all of these equations?

File: src/main/java/frc/robot/pidcontroller/PIDControllerProposed.java [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+   * Wraps error around for continuous inputs. The original error is returned if
+   * continuous mode is disabled.
+   *
+   * @param error The current error of the PID controller.
+   * @return Error for continuous inputs.
+   */
+  protected double getContinuousError(double error) {
+    if (m_continuous && m_inputRange > 0) {
+      error %= m_inputRange;
+      if (Math.abs(error) > m_inputRange / 2) {
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/pidcontroller/PIDControllerProposed.java]: What's the documentation for this if statement? Why is it being divided by two?

File: src/main/java/frc/robot/subsystems/drivetrain/DriveTrain.java [hunk 5]
```diff
@@ -19,6 +20,8 @@ public DriveTrain() {
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
+    SmartDashboard.putNumber("RobotPositionInches", getRobotPositionInches());
+    SmartDashboard.putNumber("RobotVelocityInches", getRobotVelocityInches());
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/DriveTrain.java]: Sean yelled at me, so put this in the SmartDashboard4905 class.

File: src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java [hunk 6]
```diff
@@ -42,11 +42,6 @@ public void init() {
     m_drive = new DifferentialDrive(getLeftSpeedControllerGroup(), getRightSpeedControllerGroup());
   }
 
-  @Override
-  public void periodic() {
-    // This method will be called once per scheduler run
-  }
-
   /**
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java]: Don't subsystems need to implement this?

File: src/main/java/frc/robot/subsystems/drivetrain/SparkMaxDriveTrain.java [hunk 7]
```diff
@@ -46,11 +46,6 @@ public SparkMaxDriveTrain() {
     ticksPerInch = drivetrainConfig.getInt("ticksPerInch");
   }
 
-  @Override
-  public void periodic() {
-    // This method will be called once per scheduler run
-  }
-
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/SparkMaxDriveTrain.java]: Don't subsystems need to implement this?

File: src/main/java/frc/robot/subsystems/drivetrain/TalonSRXDriveTrain.java [hunk 8]
```diff
@@ -49,14 +49,54 @@ public TalonSRXDriveTrain() {
...
+    if (m_frontRight.hasEncoder()) {
+      encoders++;
+      encoderPositionAvg += m_frontRight.getEncoderPositionTicks();
+    }
+    if (m_backRight.hasEncoder()) {
+      encoders++;
+      encoderPositionAvg += m_backRight.getEncoderPositionTicks();
+    }
+    if (encoders > 0) {
+      encoderPositionAvg = encoderPositionAvg / (ticksPerInch * encoders);
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/TalonSRXDriveTrain.java]: What's the documentation for this equation?

File: src/main/java/frc/robot/pidcontroller/PIDController4905.java (hunk 2 above)
Comment (EthanM827) [reply src/main/java/frc/robot/pidcontroller/PIDController4905.java]: the whiteboard

**Pull request merged**